### PR TITLE
Dala 2784 linked document existance

### PR DESCRIPTION
**Pull request by SiegfriedSobkowiak**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] At least 3 consecutive CI runs are successfully executed to ensure that there are no flaky tests.
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [ ] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [ ] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the clone server
  - [ ] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server

### CODE REVIEW

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/validator/DocumentExists.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+private fun callDocumentApiAndCheckFileReference(
+    fileReference: String,
+    documentControllerApi: DocumentControllerApi,
+    logger: org.slf4j.Logger,
+): Boolean {
+    try {
+        documentControllerApi.checkDocument(fileReference)
+    } catch (exception: ClientException) {
+        logger.info("The referenced document does not exist.")
+        logger.info(exception.message + exception.statusCode + exception.response)
```
Comment (jhaerDfine) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/validator/DocumentExists.kt:54]: speratoren zwischen den einzelnen nachrichten?

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/InvalidSfdrRequestTests.kt [hunk 2]
```diff
@@ -15,7 +15,7 @@ class InvalidSfdrRequestTests {
 
     fun getErrorFromApi(companyName: String): ClientException {
         val oneInvalidSfdrDataset = apiAccessor.testDataProviderForSfdrData
-            .getSpecificCompanyByNameFromPreparedFixtures(companyName)
+            .getSpecificCompanyByNameFromSfdrPreparedFixtures(companyName)
```
Comment (jhaerDfine) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/InvalidSfdrRequestTests.kt:18]: warum muss der SFDR prepared fixture angepasst werden?

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Lksg.kt [hunk 3]
```diff
@@ -1,22 +1,35 @@
...
 import org.dataland.e2etests.utils.ApiAccessor
+import org.dataland.e2etests.utils.DocumentManagerAccessor
 import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeAll
 import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.assertThrows
 
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```
Comment (jhaerDfine) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Lksg.kt:18]: warum wird das benötigt?

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Sfdr.kt [hunk 4]
```diff
@@ -1,17 +1,31 @@
...
-import org.junit.jupiter.api.Assertions
+import org.dataland.e2etests.utils.DocumentManagerAccessor
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeAll
 import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.assertThrows
 
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```
Comment (jhaerDfine) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Sfdr.kt:14]: Wieso wird das hier benötigt?

**Review by jhaerDfine — COMMENTED** [review]
filereference check funktioniert nicht für sfdr bei reports

allgemein: die tests für sfdr und lksg teilen sich viele componenten, man könnte über eine Utils klasse nachdenken

File: dataland-frontend/tests/e2e/fixtures/frameworks/lksg/LksgPreparedFixtures.ts [hunk 5]
```diff
@@ -198,3 +199,16 @@ function generateOneLksgDatasetWithOnlyNulls(): FixtureData<LksgData> {
...
   return newFixture;
 }
+
+/**
+ * Generates an LKSG dataset with a no-existing file reference.
+ * @returns the dataset
+ */
+function generateFixtureWithBrokenFileReference(): FixtureData<LksgData> {
+  const newFixture = generateLksgFixtures(1, 0)[0];
+  const dummyFileReference = "123";
```
[the remark this answers is not in the document]
Comment (brunoneumannsaavedra) [reply dataland-frontend/tests/e2e/fixtures/frameworks/lksg/LksgPreparedFixtures.ts:209]: const is called now brokenFileReference as suggested

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/validator/DocumentExists.kt (hunk 1 above)
Comment (brunoneumannsaavedra) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/validator/DocumentExists.kt]: Separatoren und Lesbarkeit von Code wurden ergänzt.

File: dataland-frontend/tests/e2e/fixtures/frameworks/lksg/LksgPreparedFixtures.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (jhaerDfine) [reply dataland-frontend/tests/e2e/fixtures/frameworks/lksg/LksgPreparedFixtures.ts:209]: image

[the remark this answers is not in the document]
Comment (brunoneumannsaavedra) [reply dataland-frontend/tests/e2e/fixtures/frameworks/lksg/LksgPreparedFixtures.ts]: You are right. The constant has been now refactored in that file as well

File: emptyCommitLoop.sh [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+# Loop to create 5 empty commits
+for ((i=1; i<=5; i++)); do
+    echo "Creating empty commit $i"
+    git commit --allow-empty -m "Empty commit $i"
+    git push origin instable-e2e-test  
+done
```
Comment (jhaerDfine) [inline emptyCommitLoop.sh:8]: dont push this file.

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/frameworks/eutaxonomynonfinancials/model/general/EutaxonomyNonFinancialsGeneral.kt [hunk 7]
```diff
@@ -28,10 +29,8 @@ data class EutaxonomyNonFinancialsGeneral(
 
     val numberOfEmployees: BigDecimal? = null,
 
-    @field:Schema(
-        implementation = Map::class,
-        example = JsonExampleFormattingConstants.REFERENCED_REPORTS_DEFAULT_VALUE,
-    )
+    @field:Schema(example = JsonExampleFormattingConstants.REFERENCED_REPORTS_DEFAULT_VALUE)
```
Comment (jhaerDfine) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/frameworks/eutaxonomynonfinancials/model/general/EutaxonomyNonFinancialsGeneral.kt:32]: wieso hast du die implementation = Map::class gelöscht?

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/frameworks/sfdr/model/general/general/SfdrGeneralGeneral.kt [hunk 8]
```diff
@@ -17,10 +18,8 @@ data class SfdrGeneralGeneral(
 
     val fiscalYearEnd: LocalDate,
 
-    @field:Schema(
-        implementation = Map::class,
-        example = JsonExampleFormattingConstants.REFERENCED_REPORTS_DEFAULT_VALUE,
-    )
+    @field:Schema(example = JsonExampleFormattingConstants.REFERENCED_REPORTS_DEFAULT_VALUE)
```
Comment (jhaerDfine) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/frameworks/sfdr/model/general/general/SfdrGeneralGeneral.kt:21]: warum hast du implementation = Map::class gelöscht?

File: dataland-frontend/tests/component/components/resources/dataTable/MultiLayerDataTableWithEuTaxonomyNonFinancials.cy.ts [hunk 9]
```diff
@@ -154,7 +154,7 @@ describe("Component test for the Eu-Taxonomy-Non-Financials view page", () => {
     const allReports = fixturesForTests.map((it) => assertDefined(it.t.general).referencedReports);
     const expectedLatestReportingPeriod = allReportingPeriods[0];
     const nameOfFirstReportOfExpectedLatestReportingPeriod = Object.keys(
-      allReports[0] as Map<string, CompanyReport>,
+      allReports[0] as Record<string, CompanyReport>,
```
Comment (jhaerDfine) [inline dataland-frontend/tests/component/components/resources/dataTable/MultiLayerDataTableWithEuTaxonomyNonFinancials.cy.ts:157]: was kann das? also record?

File: .github/workflows/CI.yaml [hunk 10]
```diff
@@ -230,6 +230,15 @@ jobs:
           DATALAND_COMMUNITY_MANAGER_CLIENT_SECRET: "password"
           DATALAND_BACKEND_CLIENT_SECRET: "password"
         run: testing/verifyOpenApiFiles.sh
+      - name: Upload OpenApi Files
```
Comment (jhaerDfine) [inline .github/workflows/CI.yaml:233]: was passiert hier?

Comment (brunoneumannsaavedra) [reply .github/workflows/CI.yaml]: Done.

**Pull request merged**